instrumentation/trace/context.go

```diff
@@ -24,6 +25,34 @@ type Context struct {
 	requestId string
 }
 
+const RequestTraceName = "X-ORBS-NAME"
+const RequestTraceTime = "X-ORBS-CREATED"
+const RequestTraceRequestId = "X-ORBS-ID"
+
+func NewFromRequest(ctx context.Context, request *http.Request) context.Context {
+	name := request.Header.Get(RequestTraceName)
+	if name != "" {
+		created := time.Now()
+		if n, err := time.Parse(time.RFC3339Nano, request.Header.Get(RequestTraceTime)); err == nil {
+			created = n
+		}
+
+		traceContex := &Context{
+			name:      name,
+			created:   created,
+			requestId: request.Header.Get(RequestTraceRequestId),
+		}
+		return PropagateContext(ctx, traceContex)
+	}
+	return ctx
+}
+
+func (c *Context) ToRequest(request *http.Request) {
```

Comment:
Consider renaming, as this name is not explanatory. Maybe `UpdateRequestWithTrace`.

Reply:
`WriteTraceToRequest`